src/cljs/rems/application.cljs

```diff
@@ -228,32 +228,44 @@
 (defn- save-attachment [{:keys [db]} [_ form-id field-id file]]
   (let [application-id (get-in db [::application :data :application/id])
         current-attachments (form/parse-attachment-ids (get-in db [::edit-application :field-values form-id field-id]))
-        description [text :t.form/upload]]
+        description [text :t.form/upload]
+        config @(rf/subscribe [:rems.config/config])
+        file-size (.. file (get "file") -size)]
     (rf/dispatch [::set-attachment-status form-id field-id :pending])
-    (post! "/api/applications/add-attachment"
-           {:url-params {:application-id application-id}
-            :body file
+    (if (some-> (:attachment-max-size config)
+                (< file-size))
+      (do
+        (rf/dispatch [::set-attachment-status form-id field-id :error])
+        (flash-message/show-default-error! :actions description
+                                           [:div
+                                            [:p [text :t.form/too-large-attachment]]
+                                            [:p [text :t.form/attachment-max-size]
+                                             ": "
+                                             (format-file-size (:attachment-max-size config))]]))
```

Comment:
We can use the `text-format` here. It's slightly better than string concatenation in that the way e.g. `:` is used here can be locale specific. One can put hiccup into the localizations too!